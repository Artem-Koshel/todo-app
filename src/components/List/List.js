import React from 'react';
import { PropTypes } from 'prop-types';
import styles from './List.module.scss';

function List({ items, Item, ControlBar }) {
  const itemsContainers = items.map((item) => (
    <li key={item.id}>
      <Item item={item} />
    </li>
  ));

  return (
    <div className={styles.list}>
      <ul>{itemsContainers}</ul>
      {ControlBar && <ControlBar />}
    </div>
  );
}

List.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    })
  ),
  Item: PropTypes.elementType.isRequired,
  ControlBar: PropTypes.elementType,
};

List.defaultProps = {
  items: [],
  ControlBar: undefined,
};

export default List;
