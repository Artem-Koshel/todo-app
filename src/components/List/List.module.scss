.list {
    margin-top: 30px;
    padding: 0;
    background-color: white;
    box-shadow: 1px 1px 50px lightgray;
    border-radius: 5px;
    
    ul {
        padding: 0;
        margin: 0;
        list-style-type: none;

        li {
            padding: 0;
            margin: 0;
            border-bottom: 1px solid lightgray;
        }
    }
}

