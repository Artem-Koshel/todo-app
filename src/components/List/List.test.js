import React from 'react';
import { render, screen } from '@testing-library/react';
import List from './List';

describe('List component', () => {
  it('renders correctly', () => {
    const items = [
      { id: 1, text: 'text 1' },
      { id: 2, text: 'text 2' },
      { id: 3, text: 'text 3' },
    ];

    const Item = ({ item }) => <span data-testid={item.id}>{item.text}</span>;
    const ControlBar = () => <div data-testid="control-bar-id" />;

    const { container } = render(<List items={items} Item={Item} ControlBar={ControlBar} />);

    expect(container).toMatchSnapshot();
  });
});
