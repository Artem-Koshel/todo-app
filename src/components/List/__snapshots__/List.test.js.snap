// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders correctly 1`] = `
<div>
  <div
    class="list"
  >
    <ul>
      <li>
        <span
          data-testid="1"
        >
          text 1
        </span>
      </li>
      <li>
        <span
          data-testid="2"
        >
          text 2
        </span>
      </li>
      <li>
        <span
          data-testid="3"
        >
          text 3
        </span>
      </li>
    </ul>
    <div
      data-testid="control-bar-id"
    />
  </div>
</div>
`;
